idaes/models_extra/power_generation/unit_models/boiler_heat_exchanger_2D.py

```diff
@@ -1320,7 +1321,8 @@ def N_Re_tube_eqn(b, t, x):
         )
         def friction_factor_tube_eqn(b, t, x):
             return (
-                b.tube_friction_factor[t, x] * b.tube_N_Re[t, x] ** 0.25
+                b.tube_friction_factor[t, x]
+                * smooth_max(b.tube_N_Re[t, x], 1, 1e-5) ** 0.25
```

Comment:
I take it these `smooth_max` terms are here to avoid IPOPT taking the Reynolds number somewhere weird (like negative numbers)?

Reply:
That's right.  The Reynolds number wants to go negative a lot resulting in a ton of evaluation errors on the way to the solution.  This makes it solve a lot cleaner and faster.

Reply:
My preferred solution would be to just bound it with `bounds=(0, None)`. There are advantages and disadvantages to both approaches. If bounds are used, step sizes are reduced but there's less nonlinearity.